prime-router/src/main/kotlin/Element.kt

```diff
@@ -1143,5 +1146,52 @@ data class Element(
                 else -> error("Internal Error: Invalid EI value '$value'")
             }
         }
+
+        /**
+         * this looks to see if there is an "all zero offset" preceded by a plus sign on the
+         * date time value. if there is, then we're going to flip this bit over to be
+         * a negative offset. Note, according to the ISO-8601 specification, UTC is *NEVER*
+         * supposed to be represented by `-0000`, only ever `+0000`. That said, RFC3339 does
+         * offer the opportunity to use `-0000` to reflect an "unknown offset time", so it
+         * is still valid and does parse. Also, Java understands and can parse from `-0000`
+         * to `+0000`, so we are not breaking our implementation there.
+         *
+         * In addition to RFC3339 allowing for `-0000`, the HL7 spec allows for that value too,
+         * so we should be good in a system that is HL7 compliant.
+         *
+         * RFC Link: https://datatracker.ietf.org/doc/html/rfc3339#section-4.3
+         */
+        fun convertPositiveOffsetToNegativeOffset(value: String): String {
+            // look for the +0 offset
+            val re = Regex(".+?\\+(00|0000|00:00)$")
```

Comment:
Perhaps move this definition as a companion object val or constant to potentially reduce regex compile performance issues.